Several not exported members from react-i18next

## Describe the bug
I followed the guide lines in the readme to enable the lib in our typescript nextjs project. I'm getting 3 errors that `NamespaceExtractor`, `WithTranslationOptions` and `Subtract` are not exported in `react-i18next/src`. Imported by `next-i18next/types.d.ts`.

Looking [here](https://github.com/i18next/react-i18next/blob/v10.11.0/src/index.d.ts) I don't see any of those types exported by `react-i18next`.

The error messages I get:
```
[0] node_modules/next-i18next/types.d.ts(5,3): error TS2305: Module '"[...path...]/node_modules/react-i18next/src"' has no exported member 'NamespaceExtractor'.
[0] node_modules/next-i18next/types.d.ts(6,3): error TS2724: Module '"[...path...]/node_modules/react-i18next/src"' has no exported member 'WithTranslationOptions'. Did you mean 'UseTranslationOptions'?
[0] node_modules/next-i18next/types.d.ts(8,3): error TS2305: Module '"[...path...]/node_modules/react-i18next/src"' has no exported member 'Subtract'.
```

## Occurs in next-i18next version
next-i18next@0.42.0

## Steps to reproduce
```
yarn add next-i18next
```

Our **server.ts** according to readme.

```typescript
import * as express from 'express';
import * as next from 'next';
import { join } from 'path';
import { parse } from 'url';
// FROM README
import nextI18NextMiddleware from 'next-i18next/middleware';

import routes from './app/routes';
// FROM README
import nextI18next from './app/utilities/i18n';

const port = parseInt(process.env.PORT!, 10) || 3000;
const dev = process.env.NODE_ENV !== 'production';
const app = next({ dev, dir: 'app' });
const handle = routes.getRequestHandler(app);

const server = express();

const rootStaticFiles: string[] = [
  '/robots.txt',
  '/sitemap.xml',
  '/favicon.ico',
];

app.prepare().then(() => {
  // FROM README
  server.use(nextI18NextMiddleware(nextI18next));

  server.get('*', (req, res) => {
    const parsedUrl = parse(req.url, true);
    if (
      parsedUrl.pathname &&
      rootStaticFiles.indexOf(parsedUrl.pathname) > -1
    ) {
      const path = join(__dirname, 'app/static', parsedUrl.pathname);
      app.serveStatic(req, res, path);
    } else {
      handle(req, res);
    }
  });

  server.listen(port);
});
```

**./app/utilities/i18n** according to readme.

```typescript
import NextI18Next from 'next-i18next';

const NextI18NextInstance = new NextI18Next({
  defaultLanguage: 'de',
  otherLanguages: [],
});

export default NextI18NextInstance;

export const {
  appWithTranslation,
  withTranslation,
} = NextI18NextInstance;
```

## Expected behaviour
Importing types which actually exist in `react-i18next`.

--- a/types.d.ts
+++ b/types.d.ts
@@ -3,7 +3,6 @@ import {
   useTranslation,
   TransProps,
   Namespace,
-  WithTranslation,
   withTranslation
 } from 'react-i18next';
 import { LinkProps } from 'next-server/link';

--- a/types.d.ts
+++ b/types.d.ts
@@ -3,10 +3,8 @@ import {
   useTranslation,
   TransProps,
   Namespace,
-  NamespaceExtractor,
-  WithTranslationOptions,
   WithTranslation,
-  Subtract
+  withTranslation
 } from 'react-i18next';
 import { LinkProps } from 'next-server/link';
 import { SingletonRouter } from 'next-server/router';
@@ -43,12 +41,7 @@ declare class NextI18Next {
 
   constructor(config: InitConfig);
 
-  withTranslation(
-    namespace: Namespace | NamespaceExtractor,
-    options?: WithTranslationOptions,
-  ): <P>(
-    component: React.ComponentType<P>,
-  ) => React.ComponentType<Subtract<P, WithTranslation>>;
+  withTranslation: typeof withTranslation;
 
   appWithTranslation<P extends object>(Component: React.ComponentType<P> | React.ElementType<P>): any;
 }

